superset/daos/tag.py

```diff
@@ -390,7 +390,12 @@ def create_tag_relationship(
         updated_tagged_objects = {
             (to_object_type(obj[0]), obj[1]) for obj in objects_to_tag
         }
-        tagged_objects_to_delete = current_tagged_objects - updated_tagged_objects
+
+        tagged_objects_to_delete = (
+            current_tagged_objects
+            if not objects_to_tag
+            else current_tagged_objects - updated_tagged_objects
+        )
```

Comment:
If `objects_to_tag` is falsy then `updated_tagged_objects` is an empty set, so this should work:

```suggestion
        tagged_objects_to_delete = current_tagged_objects - updated_tagged_objects
```

Reply:
so this logic won't work since updated tags includes both prev and new tags, so if we get a empty `objects_to_tag` we the difference logic won't catch:
```

In [1]: current_tagged_object = [1, 2, 3]

In [2]: updated_tagged_object = [1]

In [3]: updated_tagged_object = [1, 4]

In [4]: current_tagged_object - updated_tagged_object

In [5]: set(current_tagged_object) - set(updated_tagged_object)
Out[5]: {2, 3}

In [6]: updated_tagged_object = []

In [7]: set(current_tagged_object) - set(updated_tagged_object)
Out[7]: {1, 2, 3} >> won't delete anything

```